File: Level 2/할인 행사/solution.js
function solution(wants, numbers, discounts) {
  let answer = 0;
  const TOTAL_NUMBERS = 10;
  const wantsMap = {};
  wants.forEach((want, index) => {
    wantsMap[want] = numbers[index];
  });

  for (let i = 0; i + TOTAL_NUMBERS <= discounts.length; i++) {
    const discountTarget = discounts.slice(i, i + TOTAL_NUMBERS);
    const copyWantsMap = { ...wantsMap };

    discountTarget.forEach((discount) => {
      if (copyWantsMap[discount]) {
        copyWantsMap[discount]--;
      }
    });

    const isCanBuyAll =
      Object.values(copyWantsMap).reduce((acc, cur) => acc + cur, 0) === 0;

    if (isCanBuyAll) {
      answer++;
    }
  }

  return answer;
}

File: Level 2/할인 행사/solution.py
def solution(wants, numbers, discounts):
    answer = 0

    wants_dict = {}

    for index, want in enumerate(wants):
        wants_dict[want] = numbers[index]

    for i in range(len(discounts) - 9):
        discounts_10d = {}

        for discount in discounts[i : i + 10]:
            discounts_10d[discount] = discounts_10d.get(discount, 0) + 1

        if wants_dict == discounts_10d:
            answer += 1

    return answer
